packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -2197,4 +2206,102 @@ describe("typespec-client-generator-core: decorators", () => {
       strictEqual(getUsage(runner.context, Dog), UsageFlags.Output);
     });
   });
+
+  describe("@flattenProperty", () => {
```

Comment:
Should you add validation to make sure you don't create conflict?

Reply:
talked offline, a name will be made up like previously with swagger, this was the discussion of dpg teams